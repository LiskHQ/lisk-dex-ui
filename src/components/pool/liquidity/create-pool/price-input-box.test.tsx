import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { screen, fireEvent, render } from "@testing-library/react";
import PriceInputBox from "./price-input-box";

const renderComponent = () => {
  return render(<PriceInputBox defaultText={"Current LSK per ETH price"} />);
};

describe("PriceInputBox component", () => {
  it("checks if the component matches the snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("checks if input is working correctly on entering text", () => {
    const wrapper = renderComponent();
    const amountInput = wrapper.getByTestId("amount-input");
    fireEvent.change(amountInput, { target: { value: "2000" } });
    expect(screen.getByTestId("amount-input")).toHaveValue("2000");
  });
});
