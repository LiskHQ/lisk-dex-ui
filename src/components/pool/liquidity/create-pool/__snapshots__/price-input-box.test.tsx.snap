// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInputBox component checks if the component matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-amountTextTokenField-1"
  >
    <p
      class="makeStyles-inputFieldText-2"
    >
      Current 
       per 
       price
    </p>
    <input
      class="makeStyles-input-3"
      data-testid="amount-input"
    />
  </div>
</div>
`;
